### PR TITLE
Remove cli and rest_client from clkhash

### PR DESCRIPTION
**Pull request by hardbyte**

### CODE REVIEW

File: clkhash/backports.py [hunk 1]
```diff
@@ -39,11 +39,15 @@ def _int_from_bytes(bytes, byteorder, signed=False):
         hex_str = codecs.encode(bytes, 'hex')  # type: ignore
         return int(hex_str, 16)
 
-
+    try:
+        from typing import SupportsBytes
+        bytes_arg_type = Union[Iterable[int], SupportsBytes]
+    except ImportError:
+        bytes_arg_type = Iterable[int]
```
Comment (JoyceXinyueWang) [inline clkhash/backports.py]: Can you explain what this code is for? I am a bit confused here.

File: docs/development.rst [hunk 2]
```diff
@@ -23,13 +23,12 @@ Now run the unit tests and print out code coverage with `py.test`::
...
 
-Note several tests will be skipped by default. To enable the command
-line tests set the  `INCLUDE_CLI` environment variable. To enable
+Note several tests will be skipped by default. To enable
 the tests which interact with an entity service set the
 `TEST_ENTITY_SERVICE` environment variable to the target service's
 address::
 
-    $ TEST_ENTITY_SERVICE= INCLUDE_CLI= python -m pytest --cov=clkhash
+    $ TEST_ENTITY_SERVICE= python -m pytest --cov=clkhash
```
Comment (JoyceXinyueWang) [inline docs/development.rst]: Shouldn't we remove TEST_ENTITY_SERICE entirely?

Comment (hardbyte) [reply docs/development.rst]: Yup good catch, I'll search the whole project for this and remove.

File: clkhash/backports.py (hunk 1 above)
Comment (hardbyte) [reply clkhash/backports.py]: It was a (failed) attempt to make new and old versions of both Python and mypy all co-exist. I'll remove this kludge and update mypy so this should just revert back.

File: azurePipeline.yml [hunk 3]
```diff
@@ -50,13 +50,11 @@ stages:
       pythonVersions: ['3.8', '2.7']
       operatingSystems: ['ubuntu-18.04']
       architectures: ['x64']
-      fullTests: True
   - template: .azurePipeline/wholeBuild.yml  # Template reference
     parameters:
       pythonVersions: ['3.7']
       operatingSystems: ['ubuntu-18.04', 'macos-10.14', 'vs2017-win2016']
       architectures: ['x64']
```
Comment (wilko77) [inline azurePipeline.yml]: you could sort those versions into the entries below. Would be a bit neater.

File: docs/tutorials.rst [hunk 4]
```diff
@@ -1,22 +1,29 @@
...
+
+The tutorial `tutorial_api.ipynb` shows an example linkage workflow.
 
 With linkage schema version 3.0 *clkhash* introduced different comparison techniques for feature values.
 They are described in the tutorial `tutorial_comparisons.ipynb`.
 
+The notebooks can run online using binder.
+
+.. image:: https://mybinder.org/badge_logo.svg
+  :target: https://mybinder.org/v2/gh/data61/clkhash/master?filepath=docs
```
Comment (wilko77) [inline docs/tutorials.rst]: I would put this section under the `running the tutorials` heading.

File: setup.py [hunk 5]
```diff
@@ -7,16 +7,13 @@
 requirements = [
         "bashplotlib>=0.6.5",
```
[the remark this answers is not in the document]
Comment (hardbyte) [reply setup.py]: We used it in the "public" api `clkhash.describe.plot` - but I'll change that to simply return the counts.

**Pull request merged**